graphjet-core/src/main/java/com/twitter/graphjet/bipartite/MultiSegmentRandomIterator.java

```diff
@@ -121,9 +116,8 @@ private void sampleFromAliasTable() {
     }
   }
 
-  @Override
   public EdgeIterator resetForNode(long node, int numSamplesToGet, Random randomGen) {
-    super.resetForNode(node); // rebuilds the segmentEdgeRandomAccessor if needed
+    rebuildSegmentIteratorsForNode(node); // rebuilds the segmentEdgeRandomAccessor if needed
```

Comment:
in the previous super.resetForNote(node), in addition to rebuildSegmentIteratorsForNode, there are other logics such as
currentSegmentId = oldestSegmentId;
initializeCurrentSegmentIterator();

do you intend to remove them?